jib-core/src/main/java/com/google/cloud/tools/jib/tar/TarExtractor.java

```diff
@@ -73,5 +97,31 @@ public static void extract(Path source, Path destination) throws IOException {
         }
       }
     }
+    preserveModificationTimes(destination, entries, enableReproducibleTimestamps);
+  }
+
+  /**
+   * Preserve modification time of files and directories in a tar file. If a directory is not an
+   * entry in the tar file and reproducible timestamps are enabled then its modification timestamp
+   * is set to a constant value.
+   *
+   * @param destination target root for unzipping
+   * @param entries list of entries in tar file
+   * @param enableReproducibleTimestamps whether or not reproducible timestamps should be used
+   * @throws IOException when I/O error occurs
+   */
+  private static void preserveModificationTimes(
+      Path destination, List<TarArchiveEntry> entries, boolean enableReproducibleTimestamps)
+      throws IOException {
+    if (enableReproducibleTimestamps) {
+      FileTime epochPlusOne = FileTime.fromMillis(1000L);
+      new DirectoryWalker(destination)
+          .filter(Files::isDirectory)
+          .walk(path -> Files.setLastModifiedTime(path, epochPlusOne));
+    }
+    for (TarArchiveEntry entry : entries) {
+      Files.setLastModifiedTime(
```

Comment:
Oh, shouldn't we check `entry` is not a symlink? I remember setting a timestamp throws an error if it's a symlink?

Reply:
Ah it used to throw an error because calling `Files.setModifiedTime` inside the for loop on the symlink would try to change the timestamp of the target file which didn't exist/wasn't created at the time of the call.

Reply:
But hmm now that I think about it, could there be a possibility of the symlink's timestamp being set for the target?

Reply:
Oh, so it doesn't throw an error but changes the target. Adding a check seems right.

I think it's worth mentioning in the Javadoc that the timestamp of symlinks will be current even if `enableReproducibleTimestamps` is true.